elastica/modules/contact.py

```diff
@@ -23,96 +40,88 @@ class Contact:
             List of contact classes defined for rod-like objects.
     """
 
-    def __init__(self):
-        self._contacts = []
+    def __init__(self: SystemCollectionProtocol) -> None:
+        self._contacts: list[ModuleProtocol] = []
         super(Contact, self).__init__()
```

Comment:
I'm a bit confused here. What are the relation among the following three:
* Type annotation on `self` as `SystemCollectionProtocol`
* `super(Contact, self).__init__()`
* `self[i]` on line 89-90

Does the first two make mypy see the current `Contact` object as an instance of `SystemCollectionProtocol`, which allows indexing (implement `__getitem__`), so the third line is valid?

Reply:
We treat all these `modules` as a mixin to the `BaseSystemCollection`. If we don't explicitly tell `self` belongs to the bigger class as such, it doesn't recognize `__getitem__` and other variables/methods in `BaseSystemCollection`.